main.tf

```diff
@@ -98,6 +114,27 @@ resource "google_pubsub_subscription" "pull_subscriptions" {
   depends_on = [google_project_service.pubsub_api]
 }
 
+resource "google_pubsub_subscription" "bigquery_subscriptions" {
+  count                      = length(local.bigquery_subscriptions)
```

Comment:
As this is a set of new resources, can we use `for_each` instead of `count` for this? https://syedrakib.medium.com/the-problem-of-shifting-all-items-in-a-terraform-array-6b54100319c9

For the other subscriptions that are already existing, we may have to change them to a for_each approach later. But for this, as these are new, we can perhaps do it using for_each from the beginning.